php_phongo.c

```diff
@@ -507,7 +518,10 @@ zend_bool phongo_writeerror_init(zval *return_value, bson_t *bson TSRMLS_DC) /*
 		writeerror->code = bson_iter_int32(&iter);
 	}
 	if (bson_iter_init_find(&iter, bson, "errmsg") && BSON_ITER_HOLDS_UTF8(&iter)) {
-		writeerror->message = bson_iter_dup_utf8(&iter, NULL);
+		uint32_t errmsg_len;
+		const char *err_msg = bson_iter_utf8(&iter, &errmsg_len);
+
+		writeerror->message = estrndup(err_msg, errmsg_len);
```

Comment:
The `writeconcernerror->message` and `writeerror->message` assignments were longstanding bugs, which weren't apparent until I tested with libbson's default memory allocation. These strings are both freed with `efree()` in the respective class dtors.